pkg/ibmcsidriver/node.go

```diff
@@ -285,7 +286,8 @@ func (csiNS *CSINodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeSt
 	// This operation (NodeStageVolume) MUST be idempotent.
 	// If the volume corresponding to the volume_id is already staged to the staging_target_path,
 	// and is identical to the specified volume_capability the Plugin MUST reply 0 OK.
-	if device == source {
+	target, err := filepath.EvalSymlinks(source)
+	if err == nil && device == target {
```

Comment:
Can we handle err != nil ?

Reply:
If err != nil, there was some issue, in finding the softlink that source. But, that is not a blocker for us to go ahead and mount the volume. All we can do is log the error. But, just for the sake of logging the error we have to add an else condition.

Reply:
Got it!